src/Security/Member.php

```diff
@@ -1702,4 +1710,46 @@ public function getHtmlEditorConfigForCMS()
         // If can't find a suitable editor, just default to cms
         return $currentName ? $currentName : 'cms';
     }
+
+    /**
+     * This needs to be public because it's referenced in getMemberPasswordField() and called in ConfirmedPasswordField
+     */
+    public static function generateRandomPassword(array $args = []): string
+    {
+        $password = '';
+        $validator = self::password_validator();
+        $minLength = $validator ? $validator->getMinLength() : 20;
+        $maxLength = $args['maxLength'] ?? 0;
+        if ($maxLength === 0) {
+            $maxLength = max($minLength, 20);
+        } else {
+            if ($maxLength < $minLength) {
+                throw new InvalidArgumentException('maxLength argument is less than PasswordValidator minLength');
+            }
+        }
+        $dummyMember = new Member();
```

Comment:
```suggestion
        $dummyMember = Member::create();
```

Reply:
Also, _why_ do we have a dummy object? Please add a code comment explaining why we need this.

Reply:
Updated